.buildkite/pipeline.yml

```
@@ -0,0 +1,74 @@
+# Nodes with values to reuse in the pipeline.
+common_params:
+  plugins: &common_plugins
+  - &bash_cache automattic/bash-cache#v1.3.2: ~
+  env: &common_env
+    IMAGE_ID: xcode-12.5.1
+
+steps:
+  #################
+  # Build and Test
+  #################
+  - label: "🧪 Build and Test"
+    key: "test"
+    command: |
+      echo "--- :beer: Installing Dependencies"
+      brew bundle --file .buildkite/brewfile
+
+      echo "--- :rubygems: Setting up Gems"
+      install_gems
+
+      echo "--- :hammer: Build DrawText"
+      bundle exec rake compile
+
+      echo "--- :git: Setting up git-lfs"
+      git-lfs install
+
+      echo "--- :rspec: Run Rspec"
+      bundle exec rspec --profile 10 --format progress
+    env: *common_env
+    plugins: *common_plugins
+    agents:
+      queue: "mac"
+
+  #################
+  # Lint
+  #################
+  - label: "🧹 Lint"
+    command: |
+      echo "--- :rubygems: Setting up Gems"
+      install_gems
+
+      echo "--- :rubocop: Run Rubocop"
+      bundle exec rubocop
+    env: *common_env
+    plugins: *common_plugins
+    agents:
+      queue: "mac"
+
+  #################
+  # Danger
+  #################
+  - label: "⛔️ Danger"
+    command: |
+      echo "--- :rubygems: Setting up Gems"
+      install_gems
+
+      echo "--- :rubocop: Run Danger"
+      bundle exec danger
+    env: *common_env
+    plugins: *common_plugins
+    agents:
+      queue: "mac"
+
+  #################
+  # Push to RubyGems
+  #################
+  - label: ":rubygems: Publish to RubyGems"
+    key: "gem-push"
+    if: build.tag != null
+    command: .buildkite/gem-push.sh
```

Note: it's important for this step to call a separate `.sh` file (as opposed to having all the commands written inline in the `pipeline.yml` file here) because if we were to have [this line from the `.sh` script in particular](https://github.com/wordpress-mobile/release-toolkit/pull/299/files#diff-61d3706214f2ca063dd08939a472f67386bdb9f24a8a31d9698ca96aedc777baR16) be inlined in the `pipeline.yml`, then it would leak the key in clear in the `BUILDKITE_COMMAND` environment variable set by Buildkite (and be visible in the "Environment" tab of the build log)

---

Nitpick: it would be good to have this as an inline comment for future us


```suggestion
    # Note: We intentionally call a separate `.sh` script here (as opposed to having all the 
    # commands written inline) to avoid leaking a key used in the process in clear in the
    # BUILDKITE_COMMAND environment variable.
    command: .buildkite/gem-push.sh
```